ee/clickhouse/queries/funnels/funnel_trends.py

```diff
@@ -68,7 +69,11 @@ def get_step_counts_without_aggregation_query(
 
         # This is used by funnel trends when we only need data for one period, e.g. person per data point
         if specific_entrance_period_start:
-            self.params["entrance_period_start"] = specific_entrance_period_start.strftime(TIMESTAMP_FORMAT)
+            self.params["entrance_period_start"] = (
+                specific_entrance_period_start.strftime(DATE_FORMAT)
+                if interval_method in [PERIOD_TRUNC_WEEK, PERIOD_TRUNC_MONTH]
```

Comment:
Is this logic shared with somewhere else? Seems like common query or filter-level logic that could be extracted to the filter rather than here.

Can you also convince me that the fix is correct? Why is `[PERIOD_TRUNC_WEEK, PERIOD_TRUNC_MONTH]` correct? What about `PERIOD_TRUNC_DAY`?

Reply:
This is because of a quirk with `toStartOfWeek` & `toStartOfMonth` : they return a `Date`, while `toStartOfDay` returns a `DateTime`.

Thinking about it further, perhaps instead of abstracting this logic out, better to solve it in the query, by converting whatever we get to a `DateTime` itself. Will do this instead.